### PR TITLE
feat: US_17/FireBallsPower

### PR DESCRIPTION
**Pull request by GastonGutierrezC**
## Changes Made

Performed the implementation of the movement and the fireball collisions with the different components of the map, now mario can kill the enemies with the fireball.

- [x] add fireballs to Level1.json
- [x] Add the necessary system and components for the fireballs.

## Checklist

- [x] Fireballs disappear if they leave the map. 
- [x] Fireballs kill enemies.
- [x] The fireballs bounce on the floor and blocks in the sky, but when they collide with the side of a block it explodes. 

Code reviuw 

@aguirrekarina 
@OropezaHugo 
@CatCoder36

### CODE REVIEW

**Comment by OropezaHugo** [comment]
The fireballs should not be able to collition with each other

**Comment by OropezaHugo** [comment]
in random moments the fireballs disappear

**Comment by OropezaHugo** [comment]
the max fireball number in the game at the same time must be 2

**Comment by OropezaHugo** [comment]
when the fireball collides with mario after a time it kills him

File: MarioGame/Source/Components/DuctExtensionComponent.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace SuperMarioBros.Source.Components;
+
+public class DuctExtensionComponent: BaseComponent
```
Comment (CatCoder36) [inline MarioGame/Source/Components/DuctExtensionComponent.cs:3]: Why is necesary a DuckExtensionComponet for this user history?

File: MarioGame/Content/Levels/Level1.json [hunk 2]
```diff
@@ -304,6 +304,41 @@
             "position": {
                 "x": 6047,
                 "y": 544 }
+        },
+        {
+            "type": "FIREBALL",
```
Comment (CatCoder36) [inline MarioGame/Content/Levels/Level1.json:309]: fireballs as such are not entities of the level so they should not be in the level json, try to generate them based on an event or key when the level is running.

File: MarioGame/Utils/Animations.cs [hunk 3]
```diff
@@ -265,7 +273,9 @@ public static class Animations
             { EntitiesName.BLOCKERBLOCKBROWN, blockedLockAnimations},
             { EntitiesName.COIN, coin},
             { EntitiesName.DUCTCROSS , ductCross },
-            { EntitiesName.SECRETLEVELDUCTENTRANCE , ductCrossSquare }
+            { EntitiesName.SECRETLEVELDUCTENTRANCE , ductCrossSquare },
+            { EntitiesName.FIRE, fire},
+            { EntitiesName.FIREEXPROTION, fireExprotion}
```
Comment (CatCoder36) [inline MarioGame/Utils/Animations.cs:278]: Did you try to say explosion?

File: MarioGame/Source/Systems/FireBoolSystem.cs [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    if (otherEntity != null)
+    {
+        var animationComponent = otherEntity.GetComponent<AnimationComponent>();
+        if (animationComponent != null)
+        {
+            string otherEntityName = Animations.entityTextures
+                .FirstOrDefault(x => x.Value == animationComponent.animations)
+                .Key.ToString() ?? string.Empty;
+
+            if (otherEntityName == "GOOMBA" || otherEntityName == "KOOPA")
```
Comment (CatCoder36) [inline MarioGame/Source/Systems/FireBoolSystem.cs:152]: You can use the collision categories to simplify the logic of handling collisions with enemies, i.e. set up that fireball colliders can only collide with colliders of enemy type, review the property CollistionWith here:
https://nkast.github.io/Aether.Physics2D/Documentation/2.0/classnkast_1_1_aether_1_1_physics2_d_1_1_dynamics_1_1_fixture.html

**Review by CatCoder36 — COMMENTED** [review]
In general the code works but try to simplify and modularize the code, the functions should not have more than 30 lines at most or around, also try to simplify the logic or distribute it in more modules, for example the movement of a fireball if the logic becomes large put it to another system that only handles the movement and handling of appearance of the fireball that only handles the fireballSystem, remember that Aether2d gives us many functions to simplify the handling of movements and collisions, etc.

**Comment by OropezaHugo** [comment]
when I throw all the fireballs and none of them collide and go out of the camera view, I am not able to throw any fireball until I lose or win

**Review by OropezaHugo — DISMISSED** [review]

File: MarioGame/Source/Systems/FireBoolSystem.cs [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+using nkast.Aether.Physics2D.Dynamics.Contacts;
+using SuperMarioBros.Source.Components;
+using SuperMarioBros.Source.Entities;
+using SuperMarioBros.Source.Extensions;
+using SuperMarioBros.Utils.DataStructures;
+using AetherVector2 = nkast.Aether.Physics2D.Common.Vector2;
+
+namespace SuperMarioBros.Source.Systems
+{
+public class FireBoolSystem : BaseSystem
```
Comment (CatCoder36) [inline MarioGame/Source/Systems/FireBoolSystem.cs:17]: Please modularize this class better, the functions are too long.

**Review by CatCoder36 — CHANGES_REQUESTED** [review]

**Review by CatCoder36 — APPROVED** [review]

**Review by OropezaHugo — APPROVED** [review]

**Pull request merged**